### ISSUE TITLE
[8193.16] GetAssociate NWScript function failing on dominated creatures

### ISSUE THREAD

**Issue by weavejester** [Beamdog/nwn-issues#193]
**To Reproduce**

 - Dominate a creature with `EffectDominated()`:

       ApplyEffectToObject(DURATION_TYPE_PERMANENT, EffectDominated(), oCreature);

 - When using:

       object oAssociate = GetAssociate(ASSOCIATE_TYPE_DOMINATED, oMaster, i);

   it will never return `OBJECT_INVALID` no matter how high the value of `i` goes. This results in an infinite loop when iterating over dominated associates.

**Specifics**

 - Windows client, Docker server on Linux.
 - Game build: 8193.16
 - Distro platform: Steam

**Comment by Finaldeath** [comment]
A note on this: You can get more than one dominated creature, via. EffectCutsceneDominated() so it is a valid use case - some code snippits to test with the nwscript console:

```
int n = 1;
object oCreature = GetNearestObject(OBJECT_TYPE_CREATURE, OBJECT_SELF, n);
while(GetIsObjectValid(oCreature))
{
    ApplyEffectToObject(DURATION_TYPE_PERMANENT, EffectCutsceneDominated(), oCreature);
    n++;
    oCreature = GetNearestObject(OBJECT_TYPE_CREATURE, OBJECT_SELF, n);
}
```

Show associates which TMI's:
```
int n = 1;
object oAssociate = GetAssociate(ASSOCIATE_TYPE_DOMINATED, OBJECT_SELF, n);
while(GetIsObjectValid(oAssociate))
{
    SpeakString("I have dominated associate number: " + IntToString(n) + " named: " + GetName(oAssociate));
    n++;
    oAssociate = GetAssociate(ASSOCIATE_TYPE_DOMINATED, OBJECT_SELF, n);
}
```

A "fix" is to iterate the party, and check master and type, but beware using this on an NPC with their own large faction of course.

```
object oPartyMember = GetFirstFactionMember(OBJECT_SELF, FALSE);
while(GetIsObjectValid(oPartyMember))
{
    if(oPartyMember != OBJECT_SELF)
    {
        SpeakString("Party member named: " + GetName(oPartyMember) + " is associate type: " + IntToString(GetAssociateType(oPartyMember)));
    }
    oPartyMember = GetNextFactionMember(OBJECT_SELF, FALSE);
}
```

**Comment by shadguy** [comment]
Fixed in the current preview build (8193.36-6).

-Dave
